src/lib/steps/ngc.ts

```diff
@@ -40,7 +44,32 @@ export async function ngc(ngPkg: NgPackageData, basePath: string): Promise<strin
 
   // invoke ngc programmatic API
   const compilerConfig = readConfiguration(tsConfigPath);
-  performCompilation(compilerConfig);
+  const compilerResult = performCompilation(compilerConfig);
+
+  if (!compilerResult.emitResult) {
```

Comment:
Luckily, there are helper functions for this from `@angular/compiler-cli`.

```ts
  const exitCode = ng.exitCodeFromResult(result.diagnostics);
  if (exitCode === 0) {
    return Promise.resolve(
      path.resolve(ngPkg.sourcePath, tsConfig.options.outDir, tsConfig.options.flatModuleOutFile)
    );
  } else {
    return Promise.reject(new Error(
      ng.formatDiagnostics(result.diagnostics)
    ));
  }
```

If you like an early fix, I ask you to make a PR with these changes. Otherwise, let's wait for #279. I just need a few more days to correct the "write staged files to dist".

Reply:
Well, I'm not in a super hurry on that front but I did make the changes you suggested and it's much better so I just added them here. :)